script/tool/lib/src/branch_for_batch_release_command.dart

```diff
@@ -0,0 +1,311 @@
+    final String branchName = argResults!['branch'] as String;
+
+    final List<RepositoryPackage> packages = await getTargetPackages()
+        .map((PackageEnumerationEntry e) => e.package)
+        .toList();
+    if (packages.length != 1) {
+      printError('Exactly one package must be specified.');
+      throw ToolExit(2);
+    }
+    final RepositoryPackage package = packages.single;
+
+    final GitDir repository = await gitDir;
+
+    print('Parsing package "${package.displayName}"...');
+    final PendingChangelogs pendingChangelogs =
+        await _getPendingChangelogs(package);
+    if (pendingChangelogs.entries.isEmpty) {
+      print('No pending changelogs found for ${package.displayName}.');
+      return;
+    }
+
+    final Pubspec pubspec =
+        Pubspec.parse(package.pubspecFile.readAsStringSync());
+    final ReleaseInfo releaseInfo =
+        _getReleaseInfo(pendingChangelogs.entries, pubspec.version!);
+
+    if (releaseInfo.newVersion == null) {
+      print('No version change specified in pending changelogs for '
+          '${package.displayName}.');
+      return;
+    }
+
+    print('Creating and pushing release branch...');
+    await _pushBranch(
+      git: repository,
+      package: package,
+      branchName: branchName,
+      pendingChangelogFiles: pendingChangelogs.files,
+      releaseInfo: releaseInfo,
+    );
+  }
+
+  Future<PendingChangelogs> _getPendingChangelogs(
+      RepositoryPackage package) async {
+    final Directory pendingChangelogsDir =
+        package.directory.childDirectory('pending_changelogs');
+    if (!pendingChangelogsDir.existsSync()) {
+      printError(
+          'No pending_changelogs folder found for ${package.displayName}.');
+      throw ToolExit(_kExitPackageMalformed);
+    }
+    final List<File> pendingChangelogFiles = pendingChangelogsDir
+        .listSync()
+        .whereType<File>()
+        .where((File f) =>
+            f.basename.endsWith('.yaml') && f.basename != _kTemplateFileName)
+        .toList();
+    try {
+      final List<PendingChangelogEntry> entries = pendingChangelogFiles
+          .map<PendingChangelogEntry>(
+              (File f) => PendingChangelogEntry.parse(f.readAsStringSync()))
+          .toList();
+      return PendingChangelogs(entries, pendingChangelogFiles);
+    } on FormatException catch (e) {
+      printError('Malformed pending changelog file: $e');
+      throw ToolExit(_kExitPackageMalformed);
+    }
+  }
+
+  ReleaseInfo _getReleaseInfo(
+      List<PendingChangelogEntry> pendingChangelogEntries, Version oldVersion) {
+    final List<String> changelogs = <String>[];
+    int versionIndex = VersionChange.skip.index;
+    for (final PendingChangelogEntry entry in pendingChangelogEntries) {
+      changelogs.add(entry.changelog);
+      versionIndex = math.min(versionIndex, entry.version.index);
+    }
+    final VersionChange effectiveVersionChange =
+        VersionChange.values[versionIndex];
+
+    Version? newVersion;
```

Comment:
The tooling requires a new enough version of Dart that you can use expression switches:

`final Version? newVersion = switch (effectiveVersionChange) { ... };`